### PR TITLE
Add `wp user spam` command

### PR DESCRIPTION
**Pull request by BhargavBhandari90**
Command:
- wp user spam 123

For : https://github.com/wp-cli/entity-command/issues/50

### CODE REVIEW

**Comment by miya0001** [comment]
I think we shouldn't try to use SQL as much as possible.
Can you use `update_user_status()` or so?

**Comment by BhargavBhandari90** [comment]
yes, @miya0001 I just saw that function. Applying that only.

**Comment by miya0001** [comment]
Thanks @BhargavBhandari90 😄

**Comment by BhargavBhandari90** [comment]
Behat test for user.feature shows following error for `spam` command

> PHP Fatal error:  Call to undefined function get_network() in src/User_Command.php on line 1064
      Fatal error: Call to undefined function get_network() in src/User_Command.php on line 1064
      cwd: /tmp/wp-cli-test-run-user-test.feature.3-59abbe9ab397d9.99056625/
      exit status: 255

However, I am not getting this error while running command into the terminal.

So I am confused, is my code has something wrong or what?

**Comment by miya0001** [comment]
Ah, I guess we should do like following.
https://github.com/wp-cli/entity-command/blob/master/features/site.feature#L4

**Comment by BhargavBhandari90** [comment]
@miya0001 Yes, see I just did that. That was real time :stuck_out_tongue:

**Comment by miya0001** [comment]
Hi,

If `wp user spma` is passed multiple arguments and there is an error from one of them, then status code should be `1`. 
But I guess your implementation will return `0`.

We need consistency between same commands, so following is a good example to handle multiple arguments.

* https://github.com/wp-cli/extension-command/blob/master/src/Plugin_Command.php#L258-L293
* https://github.com/wp-cli/extension-command/blob/master/features/plugin-activate.feature#L15-L35

**Comment by BhargavBhandari90** [comment]
ok. Checking this.

**Comment by BhargavBhandari90** [comment]
One test is failing. because I used a function which was not available in that WP version.
How to tackle that situation?

Here is the error https://travis-ci.org/wp-cli/entity-command/jobs/271378052

**Comment by miya0001** [comment]
Do you want to use `get_network()` to mark spam the user's site too?
I guess there is a good example. :)
https://github.com/wp-cli/entity-command/blob/master/src/Site_Command.php#L749-L766

**Comment by BhargavBhandari90** [comment]
> If wp user spam is passed multiple arguments and there is an error from one of them, then status code should be 1.
But I guess your implementation will return 0.

@miya0001 As you said we need `wp user unspam` for this. So I will implement this.

**Comment by BhargavBhandari90** [comment]
Hi @miya0001 

> If wp user spam is passed multiple arguments and there is an error from one of them, then status code should be 1.

I have implemented this thing. But I don't know how to get retun code so that I can add it into feature file.

Currently I am not adding that line into `users.feature`.

**Comment by miya0001** [comment]
@BhargavBhandari90 

Please write test like following.

```
When I try `wp user spam {BBEE_ID} 9999`
Then STERR should contain:
  """
  Warning: User 9999 doesn't exist.
  """
And the return code should be 1
```

**Comment by miya0001** [comment]
Ah, sorry, it should be like following.

```
When I try `wp user spam {BBEE_ID} 9999`
Then STOUT should contain:
  """
  User {BBEE_ID} marked as spam.
  """
Then STOUT should contain:
  """
  Warning: User 9999 doesn't exist.
  """
And the return code should be 1
```

**Comment by BhargavBhandari90** [comment]
@miya0001 This should be fine now.

File: src/User_Command.php [hunk 1]
```diff
@@ -1017,4 +1020,107 @@ private static function wp_new_user_notification( $user_id, $password ) {
...
+	}
+
+	/**
+	 * Common command for updating user data.
+	 */
+	private function update_msuser_status( $user_ids, $pref, $value ) {
+
+		// If site is not multisite, then stop execution.
+		if ( ! is_multisite() ) {
+			WP_CLI::error( 'Sorry! this command is for multisite only.' );
```
Comment (schlessera) [inline src/User_Command.php]: This should mimic the other commands:
```
WP_CLI::error( 'This is not a multisite install.' );
```

**Comment by schlessera** [comment]
The grammar is of for the batch results, but this needs to be fixed in `WP_CLI\Utils`. I've opened an issue for this here: https://github.com/wp-cli/wp-cli/issues/4338

File: src/User_Command.php (hunk 1 above)
Comment (BhargavBhandari90) [reply src/User_Command.php]: Done. @schlessera

**Comment by schlessera** [comment]
Thanks, @BhargavBhandari90 !

**Pull request merged**